File: LeetCode/Java/BinaryTreeInorderTraversal.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class BinaryTreeInorderTraversal {
    List<Integer> res = new ArrayList<Integer>();
    public List<Integer> inorderTraversal(TreeNode root) {
        if (root == null) {
            return res;
        }
        if (root.left != null) {
            inorderTraversal(root.left);
        }
        res.add(root.val);
        if (root.right != null) {
            inorderTraversal(root.right);
        }
        return res;
    }
}

File: LeetCode/Go/binary_tree_inorder_traversal.go
package main

import "fmt"

type TreeNode struct {
	Val int
	Left *TreeNode
	Right *TreeNode
}

func inorderTraversal(root *TreeNode) []int {
	var nodes []int
	helper(root, &nodes)
	return nodes
}

func helper(root *TreeNode, nodes *[]int) {
	if (root == nil) {
		return
	}
	if (root.Left != nil) {
		helper(root.Left, nodes);
	}
	*nodes = append(*nodes, root.Val)
	if (root.Right != nil) {
		helper(root.Right, nodes);
	}
}

func main() {
	var root = TreeNode{1, nil, &TreeNode{2, &TreeNode{3, nil, nil}, nil}}
	var nodes = inorderTraversal(&root)
	fmt.Print(nodes)
}
